prefect_aws/s3.py

```diff
@@ -337,9 +339,118 @@ def _get_s3_client(self) -> boto3.client:
             s3_client = self.aws_credentials.get_boto3_session().client(
```

Comment:
`_resolve_path` return type annotation should actually be `str`

Should `base_path` be type `str` to match the parent?
https://github.com/PrefectHQ/prefect/blob/main/src/prefect/filesystems.py#L421-L422

Reply:
Fixed the faulty return type. I think having `basepath` typed as a `Path` makes sense because then the schema that the UI gets has more info about what type of input should be accepted.